Build fails on macOS Big Sur 11.1, arm64 (Apple Silicon)

## Build options
```
cmake -DCMAKE_EXPORT_COMPILE_COMMANDS=1 -GNinja .. && ninja
```

## Error log

```
FAILED: Truffle 
: && /Library/Developer/CommandLineTools/usr/bin/c++ -arch arm64 -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.1.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  CMakeFiles/Truffle.dir/main.cpp.o -o Truffle -F/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/System/Library/Frameworks -Wl,-rpath,/opt/homebrew/lib  -L/opt/homebrew/Cellar/sdl2/2.0.14_1/lib  -lSDL2  /opt/homebrew/lib/libSDL2_image.dylib  /opt/homebrew/lib/libSDL2_ttf.dylib  /opt/homebrew/lib/libabsl_str_format_internal.dylib  /opt/homebrew/lib/libabsl_cord.dylib  /opt/homebrew/lib/libabsl_hash.dylib  /opt/homebrew/lib/libabsl_bad_variant_access.dylib  /opt/homebrew/lib/libabsl_city.dylib  /opt/homebrew/lib/libabsl_raw_hash_set.dylib  /opt/homebrew/lib/libabsl_bad_optional_access.dylib  /opt/homebrew/lib/libabsl_hashtablez_sampler.dylib  /opt/homebrew/lib/libabsl_exponential_biased.dylib  /opt/homebrew/lib/libabsl_synchronization.dylib  /opt/homebrew/lib/libabsl_graphcycles_internal.dylib  /opt/homebrew/lib/libabsl_stacktrace.dylib  /opt/homebrew/lib/libabsl_symbolize.dylib  /opt/homebrew/lib/libabsl_malloc_internal.dylib  /opt/homebrew/lib/libabsl_debugging_internal.dylib  /opt/homebrew/lib/libabsl_demangle_internal.dylib  /opt/homebrew/lib/libabsl_time.dylib  /opt/homebrew/lib/libabsl_civil_time.dylib  /opt/homebrew/lib/libabsl_time_zone.dylib  -framework  CoreFoundation  /opt/homebrew/lib/libabsl_strings.dylib  /opt/homebrew/lib/libabsl_strings_internal.dylib  /opt/homebrew/lib/libabsl_base.dylib  /opt/homebrew/lib/libabsl_spinlock_wait.dylib  /opt/homebrew/lib/libabsl_int128.dylib  /opt/homebrew/lib/libabsl_throw_delegate.dylib  /opt/homebrew/lib/libabsl_raw_logging_internal.dylib  /opt/homebrew/lib/libabsl_log_severity.dylib && :
Undefined symbols for architecture arm64:
  "fmt::v7::detail::basic_data<void>::zero_or_powers_of_10_32_new", referenced from:
      fmt::v7::detail::count_digits(unsigned int) in main.cpp.o
  "fmt::v7::detail::basic_data<void>::zero_or_powers_of_10_64_new", referenced from:
      fmt::v7::detail::count_digits(unsigned long long) in main.cpp.o
  "fmt::v7::detail::basic_data<void>::digits", referenced from:
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned int>(char*, unsigned int, int) in main.cpp.o
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned long long>(char*, unsigned long long, int) in main.cpp.o
  "void fmt::v7::detail::vformat_to<char>(fmt::v7::detail::buffer<char>&, fmt::v7::basic_string_view<char>, fmt::v7::basic_format_args<fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<fmt::v7::type_identity<char>::type>, fmt::v7::type_identity<char>::type> >, fmt::v7::detail::locale_ref)", referenced from:
      fmt::v7::basic_format_context<fmt::v7::detail::buffer_appender<char>, char>::iterator fmt::v7::format_to<char [6], int&, 250ul, char>(fmt::v7::basic_memory_buffer<char, 250ul, std::__1::allocator<char> >&, char const (&) [6], int&) in main.cpp.o
  "fmt::v7::detail::assert_fail(char const*, int, char const*)", referenced from:
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned int>(char*, unsigned int, int) in main.cpp.o
      std::__1::make_unsigned<long>::type fmt::v7::detail::to_unsigned<long>(long) in main.cpp.o
      fmt::v7::detail::format_decimal_result<char*> fmt::v7::detail::format_decimal<char, unsigned long long>(char*, unsigned long long, int) in main.cpp.o
      fmt::v7::basic_memory_buffer<char, 250ul, std::__1::allocator<char> >::operator=(fmt::v7::basic_memory_buffer<char, 250ul, std::__1::allocator<char> >&&) in main.cpp.o
ld: symbol(s) not found for architecture arm64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
ninja: build stopped: subcommand failed.
```

## Notes

Homebrew (arm64) has no build for `abseil`.

--- a/.gitignore
+++ b/.gitignore
@@ -126,3 +126,7 @@ fabric.properties
 
 testdata/
 .idea/
+
+build/
+.clangd/
+compile_commands.json

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -9,7 +9,10 @@ find_package(SDL2_image REQUIRED)
 find_package(SDL2_ttf REQUIRED)
 find_package(SDL2 REQUIRED)
 find_package(absl REQUIRED)
+find_package(fmt REQUIRED) # used in absl internally
+find_package(spdlog REQUIRED)
 
 add_executable(Truffle main.cpp window.h non_copyable.h exception.h renderer.h dispatcher.h renderable.h texture.h logger.h scene.h scene_manager.h button.h stateful_object_manager.h font.h color.h bus.h event_publisher.h event.h)
-target_link_libraries(Truffle ${SDL2_LIBRARIES} ${SDL2_IMAGE_LIBRARIES} ${SDL2_TTF_LIBRARIES}
-         absl::strings absl::flat_hash_map absl::str_format_internal)
+target_include_directories(Truffle PRIVATE ${SPDLOG_INCLUDE_DIRS})
+target_link_libraries(Truffle PRIVATE ${SDL2_LIBRARIES} ${SDL2_IMAGE_LIBRARIES} ${SDL2_TTF_LIBRARIES} fmt::fmt
+         absl::strings absl::flat_hash_map absl::str_format)

